python/lsst/ts/m2gui/controltab/tab_default.py

```diff
@@ -116,6 +117,42 @@ def create_combo_box_ring_selection(self, callback_current_index_changed=None):
 
         return ring_selection
 
+    def create_and_start_timer(self, callback_time_out):
```

Comment:
General note: Those timer related functions so far operate for refresh timer, on self._timer. Either drop timer return/parameter and use self._timer and add _refresh_timer (which is what it now do), or even better consider using QObject.startTimer() and implement callbacks in QObject.timerEvent, which will work and do what's needed, in more Qt-centric way.

Reply:
Based on the following link:
https://doc.qt.io/qt-6/qtimer.html#details

The QTimer is a high-level programming interface for timers (this implies easier to use and maintain). It says:

An alternative to using QTimer is to call [QObject::startTimer](https://doc.qt.io/qt-6/qobject.html#startTimer)() for your object and reimplement the [QObject::timerEvent](https://doc.qt.io/qt-6/qobject.html#timerEvent)() event handler in your class (which must inherit [QObject](https://doc.qt.io/qt-6/qobject.html)). The disadvantage is that [timerEvent](https://doc.qt.io/qt-6/qtimer.html#timerEvent)() does not support such high-level features as single-shot timers or signals.

Therefore, I think you suggestion is to let the developer have more control of timer. I will consider your suggestion when I found the performance becomes an issue.

Thanks!